In [2]:
import json
import jieba

In [59]:
records = []
with open("tweet_spider_by_keyword_20230905090130.jsonl", "r") as f:
    while line := f.readline():
        records.append(json.loads(line))

with open("tweet_spider_by_keyword_20230905085828.jsonl", "r") as f:
    while line := f.readline():
        records.append(json.loads(line))

with open("tweet_spider_by_keyword_20230905204720.jsonl", "r") as f:
    while line := f.readline():
        records.append(json.loads(line))

In [3]:
print(records[0])

{'_id': '4938427672038251', 'mblogid': 'Ng7G78yf1', 'created_at': '2023-08-25 00:59:31', 'geo': None, 'ip_location': '发布于 安徽', 'reposts_count': 0, 'comments_count': 2, 'attitudes_count': 1, 'source': 'nova8Pro我由我掌镜', 'content': '#抵制日货#呼吁就呼吁 抵制就抵制，直接扫射自己人是怎么回事？谷子就吃同人谷不给IP捐钱不就行了。没看你对解决核污水有什么贡献，对人身攻击自己人倒是一套一套的。 ', 'pic_urls': ['https://wx1.sinaimg.cn/orj960/006MxfJgly1hh8bngy2o7j30s512sdkz'], 'pic_num': 1, 'isLongText': False, 'user': {'_id': '6213986410', 'avatar_hd': 'https://tvax2.sinaimg.cn/crop.0.0.1080.1080.1024/006MxfJgly8hh3j7ccru5j30u00u03za.jpg?KID=imgbed,tva&Expires=1693929689&ssig=p0IGAj1UbG', 'nick_name': '是小桉a', 'verified': False, 'mbrank': 2, 'mbtype': 2}, 'url': 'https://weibo.com/6213986410/Ng7G78yf1', 'keyword': '核污水', 'crawl_time': 1693918890}


In [4]:
content = []
for i in records:
    content.append(i["content"])

In [5]:
tokens = []
for i in content:
    tokens.extend(jieba.lcut(i))

print(len(tokens))

Building prefix dict from the default dictionary ...
Dumping model to file cache /tmp/jieba.cache
Loading model cost 0.563 seconds.
Prefix dict has been built successfully.


3493218


In [11]:
csv = [f"{i}\n" for i in tokens]
_ = ["tokens\n"]
_.extend(csv)
csv = _
csv = "".join(csv)
with open("all_tokens.csv", "w", encoding="utf-8") as f:
    f.write(csv)
    

# Time word distribution

In [60]:
content = []
for i in records:
    content.append([i["created_at"][:10], i["content"]])

for ind, i in enumerate(content):
    content[ind][1] = jieba.lcut_for_search(i[1])

In [71]:
result = dict()
for i in content:
    result[i[0]] = dict()

for i in content:
    for j in i[1]:
        try:
            result[i[0]][j] += 1
        except:
            result[i[0]][j] = 1

In [72]:
list(result["2017-02-23"].keys())

['#',
 '记者',
 '亲测',
 '福岛',
 '辐射',
 '核辐射',
 ' ',
 '在',
 '距',
 '第一',
 '核电',
 '电站',
 '核电站',
 '20',
 '公里',
 '多公里',
 '的',
 '第二',
 '附近',
 '，',
 '值',
 '是',
 '东京',
 '10',
 '倍',
 '左右',
 '。',
 '一条',
 '铁路',
 '路沿',
 '沿线',
 '铁路沿线',
 '堆积',
 '黑色',
 '塑料',
 '塑料袋',
 '里装',
 '清理',
 '出来',
 '污染',
 '土',
 '事故',
 '核事故',
 '发生',
 '后',
 '日本',
 '政府',
 '日本政府',
 '曾',
 '将',
 '周边',
 '范围',
 '内',
 '划定',
 '为',
 '警戒',
 '区域',
 '（',
 '华义',
 '）',
 'http',
 ':',
 '/',
 't',
 '.',
 'cn',
 'RihT9pY']

In [73]:
whole_dictionary = set()
for i in result:
    whole_dictionary.update(result[i].keys())
print(len(whole_dictionary))
display(whole_dictionary)

92376


{'A6OLbKeC',
 '本金',
 '玛琪雅朵',
 '东坡',
 '先看',
 '说了算',
 '大是大非',
 'A6ObpeQM',
 '权派',
 '重新处理',
 '倭排',
 '能点',
 '实物',
 '南美洲',
 '事像部',
 '与其说是',
 '球星',
 '司马昭之心',
 'A6O4T0zY',
 'policy',
 '管天',
 '鱼民',
 '旅行',
 '刷刷',
 '贩卖机',
 'YETRO',
 '出钱',
 'A6O2bjUD',
 '草淡',
 '禽流感',
 '倒影',
 '正常性',
 '司马昭',
 '归隐',
 '楠雄',
 '拖轮',
 '紅',
 'A6Oy5C0n',
 '城河',
 '倒也罢了',
 '呈上升',
 'A6OLG5It',
 '捧上去',
 '不明不白',
 '咒尽',
 '充电',
 '产期',
 '周游',
 '霸屏',
 '不见',
 '解',
 '丢脸',
 '适用性',
 'amsterdam',
 '微薄',
 'A6O4BPSk',
 '到时候',
 '靠红客',
 '衰减',
 '勇夺',
 '發表高',
 '哈德逊',
 '晞',
 '886',
 '之舞',
 '全家人',
 '从头到尾',
 '外因',
 'ǎ',
 '别索',
 '比',
 '养上',
 '一滴水',
 '周播',
 '不赢',
 'A6O4wcTE',
 '解语',
 '法不弱',
 '谢娜',
 '关键词',
 '的确',
 '起死回生',
 '游边',
 'A6O4jYrf',
 '万多元',
 '雾草',
 '干枯',
 '美事',
 '忽悠',
 '冲昏',
 '茶艺',
 'A6O40nw4',
 '天驤',
 '烦得',
 'qinlve',
 '太仓',
 '敌对势力',
 '少特',
 '吞到',
 '舆论界',
 '令人恶心',
 '我选',
 '捍然',
 '下能活',
 '猪心',
 '200911',
 '战局',
 'A6OL8jfM',
 '掩面',
 '曾以',
 '高官',
 '即刻起',
 '卖药',
 '阴凉',
 '南迁',
 '资管',
 '变迁',
 '左转',
 '乱收费',
 '机器人',
 '科学界',
 '响头',
 '工程',
 '渔政',


In [74]:
import pandas as pd
df = pd.DataFrame(columns = list(result.keys()), index=list(whole_dictionary))
display(df)

2023-08-25 2023-08-24 2023-07-04 2023-08-23 2023-08-22 2023-08-20  \
A6OLbKeC        NaN        NaN        NaN        NaN        NaN        NaN   
本金              NaN        NaN        NaN        NaN        NaN        NaN   
玛琪雅朵            NaN        NaN        NaN        NaN        NaN        NaN   
东坡              NaN        NaN        NaN        NaN        NaN        NaN   
先看              NaN        NaN        NaN        NaN        NaN        NaN   
...             ...        ...        ...        ...        ...        ...   
那核              NaN        NaN        NaN        NaN        NaN        NaN   
š               NaN        NaN        NaN        NaN        NaN        NaN   
仿照              NaN        NaN        NaN        NaN        NaN        NaN   
方圆              NaN        NaN        NaN        NaN        NaN        NaN   
CNMD            NaN        NaN        NaN        NaN        NaN        NaN   

         2022-09-04 2023-08-16 2023-07-18 2021-08-16  ... 2023-08-07  \
A6OLbKeC        NaN        NaN        NaN        NaN  ...        NaN   
本金              NaN        NaN        NaN        NaN  ...        NaN   
玛琪雅朵            NaN        NaN        NaN        NaN  ...        NaN   
东坡              NaN        NaN        NaN        NaN  ...        NaN   
先看              NaN        NaN        NaN        NaN  ...        NaN   
...             ...        ...        ...        ...  ...        ...   
那核              NaN        NaN        NaN        NaN  ...        NaN   
š               NaN        NaN        NaN        NaN  ...        NaN   
仿照              NaN        NaN        NaN        NaN  ...        NaN   
方圆              NaN        NaN        NaN        NaN  ...        NaN   
CNMD            NaN        NaN        NaN        NaN  ...        NaN   

         2023-03-12 2023-08-29 2022-12-15 2017-02-23 2022-05-03 2021-08-29  \
A6OLbKeC        NaN        NaN        NaN        NaN        NaN        NaN   
本金              NaN        NaN        NaN        NaN        NaN        NaN   
玛琪雅朵            NaN        NaN        NaN        NaN        NaN        NaN   
东坡              NaN        NaN        NaN        NaN        NaN        NaN   
先看              NaN        NaN        NaN        NaN        NaN        NaN   
...             ...        ...        ...        ...        ...        ...   
那核              NaN        NaN        NaN        NaN        NaN        NaN   
š               NaN        NaN        NaN        NaN        NaN        NaN   
仿照              NaN        NaN        NaN        NaN        NaN        NaN   
方圆              NaN        NaN        NaN        NaN        NaN        NaN   
CNMD            NaN        NaN        NaN        NaN        NaN        NaN   

         2023-08-30 2021-10-19 2023-08-31  
A6OLbKeC        NaN        NaN        NaN  
本金              NaN        NaN        NaN  
玛琪雅朵            NaN        NaN        NaN  
东坡              NaN        NaN        NaN  
先看              NaN        NaN        NaN  
...             ...        ...        ...  
那核              NaN        NaN        NaN  
š               NaN        NaN        NaN  
仿照              NaN        NaN        NaN  
方圆              NaN        NaN        NaN  
CNMD            NaN        NaN        NaN  

[92376 rows x 43 columns]

In [75]:
_counting_dict = {}
for i in whole_dictionary:
    _counting_dict[i] = 0

for i in result:
    for j_ind, j_val in result[i].items():
        _counting_dict[j_ind] = j_val
    _ = list(_counting_dict.values())
    df[i] = _
    for j in _counting_dict:
        _counting_dict[j] = 0

display(df)   

2023-08-25  2023-08-24  2023-07-04  2023-08-23  2023-08-22  \
A6OLbKeC           0           0           0           0           0   
本金                 0           0           0           0           0   
玛琪雅朵               0           0           0           0           0   
东坡                 0           0           0           0           0   
先看                 3           0           0           0           0   
...              ...         ...         ...         ...         ...   
那核                 4           0           0           0           0   
š                  1           0           0           0           0   
仿照                 0           0           0           0           0   
方圆                 0           0           0           0           0   
CNMD               0           0           0           0           0   

          2023-08-20  2022-09-04  2023-08-16  2023-07-18  2021-08-16  ...  \
A6OLbKeC           0           0           0           0           0  ...   
本金                 0           0           0           0           0  ...   
玛琪雅朵               0           0           0           0           0  ...   
东坡                 0           0           0           0           0  ...   
先看                 0           0           0           0           0  ...   
...              ...         ...         ...         ...         ...  ...   
那核                 0           0           0           0           0  ...   
š                  0           0           0           0           0  ...   
仿照                 0           0           0           0           0  ...   
方圆                 0           0           0           0           0  ...   
CNMD               0           0           0           0           0  ...   

          2023-08-07  2023-03-12  2023-08-29  2022-12-15  2017-02-23  \
A6OLbKeC           0           0           0           0           0   
本金                 0           0           1           0           0   
玛琪雅朵               0           0           0           0           0   
东坡                 0           0           0           0           0   
先看                 0           0           0           0           0   
...              ...         ...         ...         ...         ...   
那核                 0           0           5           0           0   
š                  0           0           0           0           0   
仿照                 0           0           0           0           0   
方圆                 0           0           1           0           0   
CNMD               0           0           0           0           0   

          2022-05-03  2021-08-29  2023-08-30  2021-10-19  2023-08-31  
A6OLbKeC           0           0           0           0           0  
本金                 0           0           0           0           0  
玛琪雅朵               0           0           0           0           0  
东坡                 0           0           1           0           0  
先看                 0           0           0           0           0  
...              ...         ...         ...         ...         ...  
那核                 0           0           6           0           0  
š                  0           0           0           0           0  
仿照                 0           0           1           0           0  
方圆                 0           0           0           0           0  
CNMD               0           0           0           0           0  

[92376 rows x 43 columns]

In [76]:
df.to_csv("date_token_df.csv")

In [12]:
stop_words = []
with open("stop_words.csv", "r") as f:
    while line := f.readline():
        stop_words.append(line[:-1])
print(stop_words)
stop_words.extend(["\n", ", ", " ", ".", "##", "\""])

result = {}
for i in content:
    _ = [j for j in i[1] if j not in stop_words]
    try:
        result[i[0]].extend(_)
    except:
        result[i[0]] = _

print(len(result))
print(result.keys())

['按', '按照', '俺', '俺们', '阿', '别', '别人', '别处', '别是', '别的', '别管', '别说', '不', '不仅', '不但', '不光', '不单', '不只', '不外乎', '不如', '不妨', '不尽', '不尽然', '不得', '不怕', '不惟', '不成', '不拘', '不料', '不是', '不比', '不然', '不特', '不独', '不管', '不至于', '不若', '不论', '不过', '不问', '比方', '比如', '比及', '比', '本身', '本着', '本地', '本人', '本', '巴巴', '巴', '并', '并且', '非', '彼', '彼时', '彼此', '便于', '把', '边', '鄙人', '罢了', '被', '般的', '此间', '此次', '此时', '此外', '此处', '此地', '此', '才', '才能', '朝', '朝着', '从', '从此', '从而', '除非', '除此之外', '除开', '除外', '除了', '除', '诚然', '诚如', '出来', '出于', '曾', '趁着', '趁', '处在', '乘', '冲', '等等', '等到', '等', '第', '当着', '当然', '当地', '当', '多', '多么', '多少', '对', '对于', '对待', '对方', '对比', '得', '得了', '打', '打从', '的', '的确', '的话', '但', '但凡', '但是', '大家', '大', '地', '待', '都', '到', '叮咚', '而言', '而是', '而已', '而外', '而后', '而况', '而且', '而', '尔尔', '尔后', '尔', '二来', '非独', '非特', '非徒', '非但', '否则', '反过来说', '反过来', '反而', '反之', '分别', '凡是', '凡', '', '个', '个别', '固然', '故', '故此', '故而', '果然', '果真', '各', '各个', '各位', '各种', '各自', '关于具体地说', '归齐', '归', '根据', '管', '赶', '跟', '过',

In [13]:
# Get the top 100 results
for i in result:
    result[i] = sorted(result[i])



In [ ]:
result = 
for i in result:

In [14]:
print(result["2023-08-25"][:100])

['###', '###', '#.', '#.', '#......', '&#', '&#', '&#', '&#', '&#', '&#', "'", "'", "'", "'", "'", "'", "'", "'", '++++', '++++', '++++', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',', ',']


In [8]:
for i in result:
    with open(f"date_token_{i}.json", "w", encoding="utf-8") as f:
        json.dump(result[i], f, indent=2)

{"_id": "4938427672038251", "mblogid": "Ng7G78yf1", "created_at": "2023-08-25 00:59:31", "geo": null, "ip_location": "发布于 安徽", "reposts_count": 0, "comments_count": 2, "attitudes_count": 1, "source": "nova8Pro我由我掌镜", "content": "#抵制日货#呼吁就呼吁 抵制就抵制，直接扫射自己人是怎么回事？谷子就吃同人谷不给IP捐钱不就行了。没看你对解决核污水有什么贡献，对人身攻击自己人倒是一套一套的。 ", "pic_urls": ["https://wx1.sinaimg.cn/orj960/006MxfJgly1hh8bngy2o7j30s512sdkz"], "pic_num": 1, "isLongText": false, "user": {"_id": "6213986410", "avatar_hd": "https://tvax2.sinaimg.cn/crop.0.0.1080.1080.1024/006MxfJgly8hh3j7ccru5j30u00u03za.jpg?KID=imgbed,tva&Expires=1693929689&ssig=p0IGAj1UbG", "nick_name": "是小桉a", "verified": false, "mbrank": 2, "mbtype": 2}, "url": "https://weibo.com/6213986410/Ng7G78yf1", "keyword": "核污水", "crawl_time": 1693918890}


# Flatten json

In [1]:
import numpy as np
import pandas as pd
import json
import jieba
from tqdm.notebook import tqdm

# Custom format with ANSI escape codes for dark green color
dark_green = "\033[1;32;40m"
bar_format = f"{dark_green}{{l_bar}}{{bar:50}} [{{elapsed}}]{{r_bar}}"

In [2]:
records = []
with open("tweet_spider_by_keyword_20230905090130.jsonl", "r") as f:
    while line := f.readline():
        records.append(json.loads(line))

with open("tweet_spider_by_keyword_20230905085828.jsonl", "r") as f:
    while line := f.readline():
        records.append(json.loads(line))

with open("tweet_spider_by_keyword_20230905204720.jsonl", "r") as f:
    while line := f.readline():
        records.append(json.loads(line))

In [3]:
user_table = {}
user_table_columns = list(records[0]["user"].keys())
for i in user_table_columns:
    user_table[i] = []

for i in records:
    for j in user_table_columns:
        try:
            user_table[j].append(i["user"][j])
        except:
            user_table[j].append(np.nan)

user_table = pd.DataFrame(columns=user_table_columns, data=user_table)
user_table = user_table.rename(columns={"_id": "user_id"})
user_table = user_table.set_index("user_id")
display(user_table)

avatar_hd  \
user_id                                                         
6213986410  https://tvax2.sinaimg.cn/crop.0.0.1080.1080.10...   
5634730564  https://tvax4.sinaimg.cn/crop.0.0.1080.1080.10...   
6549238750  https://tvax1.sinaimg.cn/crop.0.0.996.996.1024...   
7259760533  https://tvax3.sinaimg.cn/crop.0.0.1080.1080.10...   
1775009924  https://tvax1.sinaimg.cn/default/images/defaul...   
...                                                       ...   
5512960046  https://tva1.sinaimg.cn/crop.0.0.100.100.1024/...   
6804191525  https://tvax3.sinaimg.cn/crop.0.0.885.885.1024...   
3270195285  https://tvax3.sinaimg.cn/crop.1.0.1053.1053.10...   
1684137741  https://tvax4.sinaimg.cn/crop.0.0.1080.1080.10...   
7003696119  https://tvax2.sinaimg.cn/crop.0.0.512.512.1024...   

                           nick_name  verified  mbrank  mbtype  
user_id                                                         
6213986410                      是小桉a     False       2       2  
5634730564                       君碎霂     False       0       0  
6549238750                     苏拉卡贝特     False       0       0  
7259760533                     正宗爹味酱      True       6      12  
1775009924                     青王是心苔     False       0       0  
...                              ...       ...     ...     ...  
5512960046                   山西临猗反邪教      True       0       0  
6804191525                    俺农村来的嘞     False       0       0  
3270195285                送你一颗小福星yee     False       0       0  
1684137741                      物与我皆     False       6      12  
7003696119  daredevileeeeeeeeeeeeeee     False       1       2  

[57147 rows x 5 columns]

In [4]:
cn_prov = []
with open("Chinese_provience.csv", "r", encoding="utf-8") as f:
    prov_name = ""
    while prov_name := f.readline():
        cn_prov.append(prov_name[:-1])
display(cn_prov)

['河北',
 '山西',
 '黑龙江',
 '吉林',
 '辽宁',
 '江苏',
 '浙江',
 '安徽',
 '福建',
 '江西',
 '山东',
 '河南',
 '湖北',
 '湖南',
 '广东',
 '海南',
 '四川',
 '贵州',
 '云南',
 '陕西',
 '甘肃',
 '青海',
 '内蒙古',
 '广西',
 '西藏',
 '宁夏',
 '新疆',
 '北京',
 '天津',
 '上海',
 '重庆']

In [5]:
post_table = {}
post_table_columns = list(records[0].keys())
post_table_columns.append("country")
for i in post_table_columns:
    post_table[i] = []
#print(post_table_columns)

bar = tqdm(total=len(records), bar_format=bar_format)
for i in records:
    for j in post_table_columns:
        if j == "country":
            continue
        try:
            if j == "user":
                post_table[j].append(i["user"]["_id"])
            elif j == "ip_location":
                if i[j] is None:
                    post_table["country"].append(np.nan)
                    post_table["ip_location"].append(np.nan)
                    continue                   
                location = i[j][4:]
                if location in cn_prov:
                    post_table["country"].append("China")
                    post_table["ip_location"].append(location)
                elif location == "中国台湾":
                    post_table["country"].append("Taiwan")
                    post_table["ip_location"].append(np.nan)
                elif location == "中国香港":
                    post_table["country"].append("China")
                    post_table["ip_location"].append(location[2:])
                elif location == "中国澳门":
                    post_table["country"].append(location[2:])
                    post_table["ip_location"].append(np.nan)
                else:
                    post_table["country"].append(location)
                    post_table["ip_location"].append(np.nan)
            else:
                post_table[j].append(i[j])
        except FileExistsError:
            post_table[j].append(np.nan)
    bar.update()
bar.close()

post_table = pd.DataFrame(columns=post_table_columns, data=post_table)
post_table = post_table.rename(columns={"_id": "post_id", "user": "user_id"})
post_table = post_table.set_index("post_id")
display(post_table)

  0%|                                                   [00:00]| 0/57147 [00:00<?, ?it/s]

mblogid           created_at   geo ip_location  \
post_id                                                              
4938427672038251  Ng7G78yf1  2023-08-25 00:59:31  None          安徽   
4938442812949023  Ng84xcnaT  2023-08-25 01:59:41  None          湖南   
4938457870762097  Ng8sP3cfT  2023-08-25 02:59:31  None          江苏   
4938366619485594  Ng65DDNDs  2023-08-24 20:56:55  None          四川   
4938487734469162  Ng9eZiKDg  2023-08-25 04:58:11  None          四川   
...                     ...                  ...   ...         ...   
4938639101402406  Ngdb85SPs  2023-08-25 14:59:40  None         NaN   
4938639101399091  Ngdb85RXZ  2023-08-25 14:59:40  None          山东   
4938654155019251  Ngdzpl3JF  2023-08-25 15:59:29  None          重庆   
4938654167601014  NgdzqvTn0  2023-08-25 15:59:32  None          重庆   
4938654174153048  NgdzrhqoE  2023-08-25 15:59:33  None          山东   

                  reposts_count  comments_count  attitudes_count  \
post_id                                                            
4938427672038251              0               2                1   
4938442812949023              0               0               21   
4938457870762097              0               0               14   
4938366619485594            222             211            12062   
4938487734469162              0               2                1   
...                         ...             ...              ...   
4938639101402406              0               0                0   
4938639101399091              0               2                4   
4938654155019251              0               0                0   
4938654167601014              0               0                0   
4938654174153048              0               0                0   

                              source  \
post_id                                
4938427672038251       nova8Pro我由我掌镜   
4938442812949023           Redmi K40   
4938457870762097         荣耀70 Pro 5G   
4938366619485594                       
4938487734469162            荣耀8X Max   
...                              ...   
4938639101402406              Note 9   
4938639101399091           iPhone客户端   
4938654155019251  HUAWEI MatePad Pro   
4938654167601014    MEIZU 20 PRO热爱无界   
4938654174153048           iPhone客户端   

                                                            content  \
post_id                                                               
4938427672038251  #抵制日货#呼吁就呼吁 抵制就抵制，直接扫射自己人是怎么回事？谷子就吃同人谷不给IP捐钱不就...   
4938442812949023  #东京电力称核污水排海有问题将停止#你踏马不会起标题就别起！实际还是正在排污，未来有问题再考...   
4938457870762097         #日本核污水排海中国怎么办#全世界只有中国在反对，也只有中国做了这么多环保，，服了    
4938366619485594                                “ 你才刚射完精，就又开始扯政治了”    
4938487734469162  日本的核污水首先污染太平洋区域，进而污染全球海洋区域，随着海水蒸发，又污染全球陆地和河流，危...   
...                                                             ...   
4938639101402406    //@山西反邪教:#日本为了自己划算不惜让全世界买单##研究称日本核污水排海240天到达中国#   
4938639101399091  #日本核污水排海首日辐射值就超标#核废水污染真这么严重的话大家千万别生孩子啊！海鲜不能吃海不...   
4938654155019251               #日本向学生宣传核污水安全被骂#美化日本历史教科书的又是谁？[太开心]    
4938654167601014  就谈立场 [偷笑]//@冬亚:而且他们也没有真的讲科学，我这两天发一篇长文，什么才是讲科学。...   
4938654174153048                               差不多得了我看核污水感染的是你们的脑子    

                                                           pic_urls  pic_num  \
post_id                                                                        
4938427672038251  [https://wx1.sinaimg.cn/orj960/006MxfJgly1hh8b...        1   
4938442812949023                                                 []        0   
4938457870762097                                                 []        0   
4938366619485594  [https://wx1.sinaimg.cn/orj960/007VjdlPgy1hh84...        1   
4938487734469162                                                 []        0   
...                                                             ...      ...   
4938639101402406                                                 []        0   
4938639101399091                                           

In [6]:
token_table = {"post_id": [], "token": []}
stop_words = []
with open("stop_words.csv", "r") as f:
    while line := f.readline():
        stop_words.append(line[:-1])
stop_words.extend(["\n", " ", "\""])

bar = tqdm(total=len(post_table.index), bar_format=bar_format)
for row_index, post in post_table.iterrows():
    token_list = jieba.lcut_for_search(post[8])
    for token in token_list:
        if token not in stop_words:
            token_table["post_id"].append(row_index)
            token_table["token"].append(token)
    bar.update()
bar.close()

token_table = pd.DataFrame(data=token_table)
display(token_table)

  0%|                                                   [00:00]| 0/57147 [00:00<?, ?it/s]

/tmp/ipykernel_1117707/3960783883.py:10: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  token_list = jieba.lcut_for_search(post[8])
Building prefix dict from the default dictionary ...
Loading model from cache /tmp/jieba.cache
Loading model cost 0.864 seconds.
Prefix dict has been built successfully.


post_id token
0        4938427672038251    抵制
1        4938427672038251    日货
2        4938427672038251    呼吁
3        4938427672038251    呼吁
4        4938427672038251    抵制
...                   ...   ...
2208143  4938654174153048   差不多
2208144  4938654174153048    看核
2208145  4938654174153048    污水
2208146  4938654174153048    感染
2208147  4938654174153048    脑子

[2208148 rows x 2 columns]

In [30]:
with pd.ExcelWriter(
    "table.xlsx",
    mode="w",
    engine="openpyxl",
    #if_sheet_exists="replace",
) as writer:
    post_table.to_excel(writer, sheet_name="posts")
    user_table.to_excel(writer, sheet_name="users")

In [17]:
from argostranslate import translate
from argostranslate import package

package.install_from_path('translate-zh_en-1_7.argosmodel')

def get_argos_model(source, target):
    lang = f'{source} -> {target}'
    source_lang = [model for model in translate.get_installed_languages() if lang in map(repr, model.translations_from)]
    print(source_lang)
    target_lang = [model for model in translate.get_installed_languages() if lang in map(repr, model.translations_to)]
    print(target_lang)
    
    return source_lang[0].get_translation(target_lang[0])

argos_zh_en = get_argos_model('Chinese', 'English')

/home/loe/.python_module/requests/__init__.py:109: RequestsDependencyWarning: urllib3 (1.26.13) or chardet (None)/charset_normalizer (3.2.0) doesn't match a supported version!
  warnings.warn(


In [23]:
# Processed Token Table
processed_token_table = {}
for i in token_table["token"]:
    try:
        processed_token_table[i] += 1
    except:
        processed_token_table[i] = 1

_processed_token_table = {"Token": [], "Count": [], "Token_en": []}
bar = tqdm(total=len(processed_token_table))
for ind, (key, value) in enumerate(processed_token_table.items()):
    if ind >= 1000:
        break
    if value <= 150:
        bar.update()
        continue
    _processed_token_table["Token"].append(key)
    _processed_token_table["Token_en"].append(argos_zh_en.translate(key))
    _processed_token_table["Count"].append(value)
    bar.update()
bar.close()

display(len(_processed_token_table["Token"]))

_processed_token_table_df = pd.DataFrame(data = _processed_token_table)
_processed_token_table_df.to_csv("processed_token_table.csv")
with open("processed_token_table.json", "w", encoding="utf-8") as f:
    json.dump(_processed_token_table, f, indent=2)

  0%|          | 0/91612 [00:00<?, ?it/s]

572

In [7]:
token_table.to_json("token_table.json")